recipes/cpr/all/conanfile.py

```diff
@@ -101,24 +99,25 @@ def validate(self):
         if not self._uses_valid_abi_and_compiler:
             raise ConanInvalidConfiguration(f"Cannot compile {self.ref} with libstdc++ on clang < 9")
 
-        ssl_library = str(self.options.with_ssl)
-        
-        if ssl_library == "openssl" and is_apple_os(self):
-            raise ConanInvalidConfiguration("OpenSSL is not supported on macOS")
-        if ssl_library == "darwinssl" and not is_apple_os(self):
-            raise ConanInvalidConfiguration("DarwinSSL is only supported on macOS")
-        if ssl_library == "winssl" and self.settings.os != "Windows":
-            raise ConanInvalidConfiguration("WinSSL is only on Windows")
+        if self.options.with_ssl:
+            ssl_library = str(self.options.with_ssl)
 
-        if ssl_library not in (CprConan._AUTO_SSL, CprConan._NO_SSL, "winssl") and ssl_library != self.dependencies["libcurl"].options.with_ssl:
-            raise ConanInvalidConfiguration(
-                f"{self.ref}:with_ssl={self.options.with_ssl} requires libcurl:with_ssl={self.options.with_ssl}"
-            )
+            if ssl_library == "openssl" and is_apple_os(self):
+                raise ConanInvalidConfiguration("OpenSSL is not supported on macOS")
+            if ssl_library == "darwinssl" and not is_apple_os(self):
+                raise ConanInvalidConfiguration("DarwinSSL is only supported on macOS")
+            if ssl_library == "winssl" and self.settings.os != "Windows":
+                raise ConanInvalidConfiguration("WinSSL is only on Windows")
 
-        if ssl_library == "winssl" and self.dependencies["libcurl"].options.with_ssl != "schannel":
-            raise ConanInvalidConfiguration(
-                f"{self.ref}:with_ssl=winssl requires libcurl:with_ssl=schannel"
-            )
+            if ssl_library != "winssl" and ssl_library != self.dependencies["libcurl"].options.with_ssl:
```

Comment:
```suggestion
            if ssl_library != "winssl" and ssl_library != self.dependencies["libcurl"].options.with_ssl:
```

You don't want to do this check when `self.options.with_ssl=False`